inspirehep/modules/workflows/tasks/refextract.py

```diff
@@ -184,3 +189,60 @@ def extract_references_from_raw_ref(reference, custom_kbs_file=None):
     return extract_references_from_text(
         raw_ref['value'], source=raw_ref['source'], custom_kbs_file=custom_kbs_file
     )
+
+
+def match_references(references):
+    """Match references to record ids given a list of references.
+
+    Args:
+        references: Schema compliant list of references
+
+    Returns:
+        associated_references: The same list with the associated record ids (if any)
+    """
+
+    for i, reference in enumerate(references):
+        matched_recid = match_reference(reference)
+        if matched_recid:
+            references[i]['record'] = get_record_ref(matched_recid, 'literature')
+
+    return references
+
+
+def match_reference(reference):
+    """Match references given a reference metadata using InspireMatcher queires.
+
+    Args:
+        reference: The reference metadata
+        config: The configurtaion(s) for InspireMatcher queries
+
+    Returns:
+        The record ID of the matched reference
+    """
+
+    config_default = current_app.config['WORKFLOWS_REFERENCE_MATCHER_DEFAULT_CONFIG']
+    config_jcap_and_jhep = current_app.config['WORKFLOWS_REFERENCE_MATCHER_JHEP_AND_JCAP_CONFIG']
+    config_data = current_app.config['WORKFLOWS_REFERENCE_MATCHER_DATA_CONFIG']
```

Comment:
Let's do one thing at a time: if we were to match on data with the current code then we will produce the wrong `$ref`. So we can't have this code in here yet, unless we solve that other issue first.

Reply:
To be clear: this applies to the current PR, but by the end of the sprint we **are** going to handle data references!

Reply:
Do we handle the data issue as well in this PR, or should we try to merge this before doing that?

Reply:
I'd say that we limit the scope of the current PR to Literature references, but in the very next PR we're going to handle Data references as well.